eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+    count_corr_novel = str(metadata_count_correct["novel"])
+
+    file_behav.write(f"{count_corr_standard}\t{count_corr_novel}\t{count_corr_target}")
+
+    file_behav.write("\n\nResponse_mean, Response_std\n")
+    file_behav.write(str(response_mean) + "\t" + str(response_std))
+
+    file_behav.write("\n\nd'\n")
+    file_behav.write(
+        str(SDT(num_Hits, num_Misses, num_FalseAlarms, num_CorrectRejections)["d"])
+    )
+
+    file_behav.close()  # to change file access modes
+
+
+def epoching(raw, events, event_id, metadata):
+    """Epoching.
+    Parameters
+    ----------
+    %(raw)s
+    %(events)s
+    %(event_id)s
+    %(metadata)s
+    """
+    epochs_tmin, epochs_tmax = -0.2, 0.8
+    epochs = Epochs(
+        raw=raw,
+        tmin=epochs_tmin,
+        tmax=epochs_tmax,
+        events=events,
+        event_id=event_id,
+        metadata=metadata,
+        reject=None,
+        preload=True,
+        baseline=(None, 0),
+        picks="eeg",
+    )
+    return epochs
+
+def get_rejection(epochs):
+    starttime = time.time()
+    reject = get_rejection_threshold(epochs, decim=1, ch_types="eeg", random_state=888)
+    endtime = time.time()
+    diff = endtime - starttime
+    minutes = str(int(diff // 60)).zfill(2)
+    seconds = str(int(diff % 60)).zfill(2)
+    print("\nPeak-to-peak rejection threshold computed: %s", reject)
+    print(f"Elapsed {minutes}:{seconds}\n")
+    return reject
+
+def clean_epochs_from_rejection(epochs, reject, FNAME_STEM, DERIVATIVES_SUBFOLDER):
+    # reject = dict(eeg=100e-6,      # unit: V (EEG channels)
+    #                  # unit: V (EOG channels)
+    #               )
+
+    stim_before = [el[2] for el in epochs.events]
+    count_stim_before = Counter(stim_before)
+
+
+
+
+    epochs.drop_bad(reject=reject)
+
+    stim_after = [el[2] for el in epochs.events]
+    count_stim_after = Counter(stim_after)
+
+    data = [['1', count_stim_before[1]-count_stim_after[1]], 
+            ['2', count_stim_before[2]-count_stim_after[2]], 
+            ['3', count_stim_before[3]-count_stim_after[3]]] 
+
+    df_count = pd.DataFrame(data, columns=['Stim', 'nb_dropped']) 
+
+    df_count.to_csv(DERIVATIVES_SUBFOLDER / (FNAME_STEM + "_step9_c2-drop-epochs.csv"))
+
+    fig = epochs.plot_drop_log(subject=FNAME_STEM)
+
+    FNAME_DROP_LOG = (
+        DERIVATIVES_SUBFOLDER / "plots" / (FNAME_STEM + "_step9_epochs-rejected.svg")
+    )
+    fig.savefig(FNAME_DROP_LOG, transparent=True)
+
+    totals = Counter(i for i in list(itertools.chain.from_iterable(epochs.drop_log)))
+    df_drops = pd.DataFrame.from_dict(totals, orient = "index")
+    df_drops = df_drops.rename(columns={0: FNAME_STEM})
+    df_drops = df_drops.sort_values(by=[FNAME_STEM], ascending = False)
+    df_drops.to_csv(DERIVATIVES_SUBFOLDER / (FNAME_STEM + "_step9_c3-drop-channel-log.csv"))
+
+    return(epochs)
+
+def save_evoked(epochs, event_id, FNAME_STEM, DERIVATIVES_SUBFOLDER):
```

Comment:
More capital letter-constants where they should be simple variables in lower case ;)